vsphere/resource_vsphere_entity_permissions.go

```diff
@@ -221,3 +228,29 @@ func resourceVSphereEntityPermissionsCustomizeDiff(d *schema.ResourceDiff, meta
 	}
 	return nil
 }
+
+func permissionsDiffSuppressFunc(k, old, new string, d *schema.ResourceData) bool {
+	oldPermissions, newPermissions := d.GetChange("permissions")
+	oldPermissionsArr := oldPermissions.([]interface{})
+	newPermissionsArr := newPermissions.([]interface{})
+	if len(oldPermissionsArr) != len(newPermissionsArr) {
+		return false
+	}
+	sort.Slice(newPermissionsArr[:], func(i, j int) bool {
+		return strings.ToLower(newPermissionsArr[i].(map[string]interface{})["user_or_group"].(string)) <
+			strings.ToLower(newPermissionsArr[j].(map[string]interface{})["user_or_group"].(string))
+	})
+	for i := 0; i < len(oldPermissionsArr); i++ {
+		if strings.ToLower(oldPermissionsArr[i].(map[string]interface{})["user_or_group"].(string)) !=
+			strings.ToLower(newPermissionsArr[i].(map[string]interface{})["user_or_group"].(string)) ||
+			oldPermissionsArr[i].(map[string]interface{})["role_id"].(string) !=
+				newPermissionsArr[i].(map[string]interface{})["role_id"].(string) ||
+			oldPermissionsArr[i].(map[string]interface{})["is_group"].(bool) !=
+				newPermissionsArr[i].(map[string]interface{})["is_group"].(bool) ||
+			oldPermissionsArr[i].(map[string]interface{})["propagate"].(bool) !=
+				newPermissionsArr[i].(map[string]interface{})["propagate"].(bool) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Thanks for adding this function. I think there is one more case we need to account for, let me know if this makes sense to you:

* User on the current version of the provider creates permissions but does not order them in their config.
* Once they upgrade we will sort the object and store it in the state.
* Next time a plan runs the diff suppress func will be evaluated. The "old" value is the unsorted one, and the "new" value is the sorted. This will produce a diff and ask the user to apply again.

My suggestion would be to sort both lists (old and new) and then use reflect.DeelEqual (https://golang.org/pkg/reflect/#DeepEqual) to decide if the two lists are equal or not.

What do you think?

Reply:
makes sense!! I will update the code.